### ISSUE TITLE
Failed to read cluster ca certificate when joining worker node

### ISSUE THREAD

**Issue by Ed87** [k0sproject/k0s#1437]
0

I am attempting to setup a multi-node k8s cluster as per this  [,](https://techviewleo.com/deploy-kubernetes-cluster-on-debian-using-k0s/) but I face the error below when I try to join one of the nodes to the master node :

`k0s token create --role=worker
    WARN[2022-01-12 13:55:31] no config file given, using defaults
    Error: failed to read cluster ca certificate from /var/lib/k0s/pki/ca.crt: open 
    /var/lib/k0s/pki/ca.crt: no such file or directory. check if the control plane is 
    initialized on this node`

I verified from the control node that this file does, exist however :
`cd /var/lib/k0s/pki
ls`

I am new to k8s setup from scratch , and a bit unsure which configuration item I need to fix (and where between master and worker nodes?). My research shows that **_A token is required when joining a new worker node to the Kubernetes cluster -- This token is generated from the control node_**. It also says that **_When you bootstrap a cluster with kubeadm, a token is generated which expires after 24 hours_**

When I try to check for the existence of a token on master node I get :

` kubeadm tokens list
-bash: kubeadm: command not found`

I am unsure however if this is correct ( Is k0s even bootstrapped with kubeadm ?? ).

However using k0s CLI syntax I can see that they are no tokens on the master:
`k0s token list
No k0s join tokens found`

`My question :

1.What do I need to do for the file that is reporting as missing.
2. Is this error (possibly) related to the issue of token above and if so do I first need to regenerate the token at the master node end.
3.I have encountered multiple fixes at this Github Issue but I am not sure if any of them applies to my issue. Last thing I need is to break the current setup before I have even finished the cluster setup.`

Environment 
Master node : Debian 10 Buster 
Worker node : Debian 10 Buster

**Comment by jnummelin** [comment]
Are you running the command `k0s token create ...` with elevated permissions, e.g. as root or via `sudo`? The certs obviously have restricted permissions which would explain the `no such file or directory` error

**Comment by Ed87** [comment]
@jnummelin  I am logged in as root. Regardless I have retried using :

`sudo k0s token create --role=worker`

and I still get the same exception.

**Comment by jnummelin** [comment]
What permissions you see for the files in `/var/lib/k0s/pki/`?

**Comment by Ed87** [comment]
I have managed to join the node. Turns out I was executing the command from the worker node instead of the master node. I also ran this command by mistake on the worker node -

`k0s install controller --enable-worker`

Will this not affect my setup or alternatively how can I undo this action. When I ran the command however I diddnt get the expected output which (might) suggest the mistake was handled internally by k0s . The expected output (as per this [is:](https://techviewleo.com/deploy-kubernetes-cluster-on-debian-using-k0s/) :
 `INFO[2021-11-21 22:36:55] no config file given, using defaults         
INFO[2021-11-21 22:36:55] creating user: etcd                          
INFO[2021-11-21 22:36:55] creating user: kube-apiserver                
INFO[2021-11-21 22:36:55] creating user: konnectivity-server           
INFO[2021-11-21 22:36:55] creating user: kube-scheduler                
INFO[2021-11-21 22:36:56] Installing k0s service `

On another note can I run Docker workloads on the cluster, if not, how can I configure k0s to use Docker as the container runtime.
